nrf-hal-common/src/comp.rs

```diff
@@ -0,0 +1,281 @@
+            Transition::Cross => w.cross().set_bit(),
+            Transition::Down => w.down().set_bit(),
+            Transition::Up => w.up().set_bit(),
+        });
+        self
+    }
+
+    /// Enables the comparator and waits until it's ready to use.
+    #[inline(always)]
+    pub fn enable(&self) {
+        self.comp.enable.write(|w| w.enable().enabled());
+        self.comp.tasks_start.write(|w| unsafe { w.bits(1) });
+        while self.comp.events_ready.read().bits() == 0 {}
+    }
+
+    /// Disables the comparator.
+    #[inline(always)]
+    pub fn disable(&self) {
+        self.comp.tasks_stop.write(|w| unsafe { w.bits(1) });
+        self.comp.enable.write(|w| w.enable().disabled());
+    }
+
+    /// Checks if the `Up` transition event has been triggered.
+    #[inline(always)]
+    pub fn is_up(&self) -> bool {
+        self.comp.events_up.read().bits() != 0
+    }
+
+    /// Checks if the `Down` transition event has been triggered.
+    #[inline(always)]
+    pub fn is_down(&self) -> bool {
+        self.comp.events_down.read().bits() != 0
+    }
+
+    /// Checks if the `Cross` transition event has been triggered.
+    #[inline(always)]
+    pub fn is_cross(&self) -> bool {
+        self.comp.events_cross.read().bits() != 0
+    }
+
+    /// Returns reference to `Up` transition event endpoint for PPI.
+    #[inline(always)]
+    pub fn event_up(&self) -> &Reg<u32, _EVENTS_UP> {
+        &self.comp.events_up
+    }
+
+    /// Returns reference to `Down` transition event endpoint for PPI.
+    #[inline(always)]
+    pub fn event_down(&self) -> &Reg<u32, _EVENTS_DOWN> {
+        &self.comp.events_down
+    }
+
+    /// Returns reference to `Cross` transition event endpoint for PPI.
+    #[inline(always)]
+    pub fn event_cross(&self) -> &Reg<u32, _EVENTS_CROSS> {
+        &self.comp.events_cross
+    }
+
+    /// Marks event as handled.
+    #[inline(always)]
+    pub fn reset_event(&self, event: Transition) {
+        match event {
+            Transition::Cross => self.comp.events_cross.write(|w| w),
+            Transition::Down => self.comp.events_down.write(|w| w),
+            Transition::Up => self.comp.events_up.write(|w| w),
+        }
+    }
+
+    /// Marks all events as handled.
+    #[inline(always)]
+    pub fn reset_events(&self) {
+        self.comp.events_cross.write(|w| w);
+        self.comp.events_down.write(|w| w);
+        self.comp.events_up.write(|w| w);
+    }
+
+    /// Returns the output state of the comparator.
+    #[inline(always)]
+    pub fn read(&self) -> CompResult {
+        self.comp.tasks_sample.write(|w| unsafe { w.bits(1) });
+        match self.comp.result.read().result().is_above() {
+            true => CompResult::Above,
+            false => CompResult::Below,
+        }
+    }
+
+    /// Consumes `self` and returns back the raw `COMP` peripheral.
+    #[inline(always)]
+    pub fn free(self) -> COMP {
+        self.comp
+    }
+}
+
+pub enum OperationMode {
```

Comment:
Probably want to hit all of these enums with derives for Debug, Eq, PartialEq, Clone, and Copy.